packages/now-go/index.ts

```diff
@@ -49,16 +50,22 @@ export async function build({
     getWriteableDirectory(),
   ]);
 
+  const srcPath = join(goPath, 'src', 'lambda');
+  let downloadedFiles = await download(files, srcPath, meta);
```

Comment:
This still ends up copying the files to a temp directory, which kind of defeats the speed optimizations that the `cwd === workPath` change introduces.

I think it needs to be more like:

```suggestion
  let downloadedFiles = await download(files, workPath, meta);
```

As far as the `.now` directory goes, how about we run the analyze go program after this part on the entrypoint to get the list of relevant files related to the entrypoint file, and then copy _only_ those files into `${workPath}/.now/cache/${entrypoint}/src/lambda`. Moving the `goPath` into there has the advantage that subsequent boots of `now dev` on the project will already have the Go dependencies installed which will make builds faster.

Reply:
I updated it to work like you mention. right now we parsed the entrypoint first, then only copy relevant files to work with. Also, handle subdirectory properly as well.

I tried to keep all the changes only related to dev only. 

Let me know what do you think? 🙏🙇‍♂️